src/test/java/com/alvarium/utils/EncoderTest.java

```diff
@@ -0,0 +1,19 @@
+package com.alvarium.utils;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+import static org.junit.Assert.assertEquals;
+import org.junit.Test;
+
+public class EncoderTest {
+  @Test
+  public void bytesToHexShouldReturnDoubleLength() throws NoSuchAlgorithmException{
+    for (int i = 0; i < 20; i++) {
+      byte[] data = new byte[i];
+      SecureRandom.getInstanceStrong().nextBytes(data);
```

Comment:
The above line causes a build error for me and I'm curious if you can duplicate after rebasing this branch from `main`.

The error is fixed for me by by adding the following to the pom.xml. Please confirm.
```
<plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <source>11</source>
+          <target>11</target>
+        </configuration>
+      </plugin>
+    </plugins>

```

Reply:
rebased and fixed in e44da0996543e7efa62ecb549cbe49a0700f1f82. I've also added further unit tests.